tests/components/camera/test_init.py

```diff
@@ -1005,3 +1005,53 @@ async def test_webrtc_provider_not_added_for_native_webrtc(
     assert camera_obj._webrtc_provider is None
     assert camera_obj._supports_native_sync_webrtc is not expect_native_async_webrtc
     assert camera_obj._supports_native_async_webrtc is expect_native_async_webrtc
+
+
+@pytest.mark.usefixtures("mock_camera", "mock_stream_source")
+async def test_camera_capabilities_changing_non_native_support(
+    hass: HomeAssistant,
+    hass_ws_client: WebSocketGenerator,
+    mock_test_webrtc_cameras: list[camera.Camera],
```

Comment:
Isn't it here that we should remove `mock_test_webrtc_cameras` since we want to test non native (although we use the demo camera entity)?

Reply:
here too